### PR TITLE
Add MySQL Flex API Version 2021-12-01-preview

### PR DESCRIPTION
**Pull request by zehao-ms**
<i>MSFT employees can try out our new experience at <b>[OpenAPI Hub](https://aka.ms/openapiportal) </b> - one location for using our validation tools and finding your workflow. 
</i>

### Changelog
Add a changelog entry for this PR by answering the following questions:
  1. What's the purpose of the update?
      - [ ] new service onboarding
      - [x] new API version
      - [ ] update existing version for new feature
      - [ ] update existing version to fix swagger quality issue in s360
      - [ ] Other, please clarify
  2. When are you targeting to deploy the new service/feature to public regions? Please provide the date or, if the date is not yet available, the month.  June
  3. When do you expect to publish the swagger? Please provide date or, the the date is not yet available, the month. June
  4. If updating an existing version, please select the specific language SDKs and CLIs that must be refreshed after the swagger is published.
      - [x] SDK of .NET (need service team to ensure code readiness)
      - [x] SDK of Python
      - [x] SDK of Java
      - [x] SDK of Js
      - [x] SDK of Go
      - [ ] PowerShell
      - [ ] CLI
      - [ ] Terraform
      - [ ] No refresh required for updates in this PR

### Contribution checklist:
- [ ] I commit to follow the [Breaking Change Policy](http://aka.ms/bcforapi) of "no breaking changes"
- [ ] I have reviewed the [documentation](https://aka.ms/ameonboard) for the workflow.
- [ ] [Validation tools](https://aka.ms/swaggertools) were run on swagger spec(s) and errors have all been fixed in this PR. [How to fix?](https://aka.ms/ci-fix)

If any further question about AME onboarding or validation tools, please view the [FAQ](https://aka.ms/faqinprreview).

### ARM API Review Checklist 

> **Applicability**: :warning: 
>
> If your changes encompass only the following scenarios, you should SKIP this section, as these scenarios do not require ARM review.
> - Change to data plane APIs 
> - Adding new properties 
> - All removals

Otherwise your PR may be subject to ARM review requirements. Complete the following:
- [x] Check this box if any of the following appy to the PR so that the label "ARMReview" and "WaitForARMFeedback" will be added by bot to kick off ARM API Review.  Missing to check this box in the following scenario may result in delays to the ARM manifest review and deployment.
  - Adding a new service
  - Adding new API(s)
  - Adding a new API version
    -[ ] To review changes efficiently, ensure you are using OpenAPIHub to initialize the PR for adding a new version. More details, refer to the [wiki](https://dev.azure.com/azure-sdk/internal/_wiki/wikis/internal.wiki/208/OpenAPI-Hub-Adding-new-API-version).

- [x] Ensure you've reviewed following [guidelines](https://aka.ms/rpguidelines) including [ARM resource provider contract](https://github.com/Azure/azure-resource-manager-rpc) and [REST guidelines](https://github.com/microsoft/api-guidelines/blob/vNext/azure/Guidelines.md). Estimated time (4 hours). This is required before you can request review from ARM API Review board.

- [ ] If you are blocked on ARM review and want to get the PR merged with urgency, please get the ARM oncall for reviews (*RP Manifest Approvers* team under <ins>Azure Resource Manager service</ins>) from IcM and reach out to them. 

### Breaking Change Review Checklist 
If any of the following scenarios apply to the PR, request approval from the Breaking Change Review Board as defined in the [Breaking Change Policy](http://aka.ms/bcforapi). 
- [ ] Removing API(s) in a stable version
- [ ] Removing properties in a stable version
- [ ] Removing API version(s) in a stable version
- [ ] Updating API in a stable or public preview version with Breaking Change Validation errors
- [ ] Updating API(s) in public preview over 1 year (refer to [Retirement of Previews](https://dev.azure.com/msazure/AzureWiki/_wiki/wikis/AzureWiki.wiki/37683/Retirement-of-Previews))

**Action**: to initiate an evaluation of the breaking change, create a new intake using the [template for breaking changes](https://aka.ms/Breakingchangetemplate). Addition details on the process and office hours are on the [Breaking change Wiki](https://dev.azure.com/msazure/AzureWiki/_wiki/wikis/AzureWiki.wiki/37684/Breaking-Changes).

Please follow the link to find more details on [PR review process](https://aka.ms/SwaggerPRReview).

### CODE REVIEW

**Comment by zehao-ms** [comment]
Hi @lirenhe , can you please help review this PR?
We added a new API version for flexible MySQL service. Thank you

**Comment by lirenhe** [comment]
> Hi @lirenhe , can you please help review this PR? We added a new API version for flexible MySQL service. Thank you

This PR requires ARM team for review.  Could you work with them on that? Thanks

**Comment by msyyc** [comment]
updated python configuration

**Comment by lirenhe** [comment]
> Hi @msyyc , thanks for your help. Could you please help merge this PR?

This PR is not ready to merge yet. Please work with ARM team to get it reviewed.  Thanks

**Comment by zehao-ms** [comment]
> > Hi @msyyc , thanks for your help. Could you please help merge this PR?
> 
> This PR is not ready to merge yet. Please work with ARM team to get it reviewed. Thanks

Thank you @lirenhe , @msyyc Is there anything left we need to modify? Or who else should we ask for review? Thanks.

**Comment by msyyc** [comment]
Hi @zehao-ms Python approved for the PR. But as https://github.com/Azure/azure-rest-api-specs/pull/19217#issuecomment-1143074945 said, it needs ARM review before merge

File: specification/mysql/resource-manager/Microsoft.DBforMySQL/preview/2021-12-01-preview/FlexibleServers.json [hunk 1]
```diff
@@ -0,0 +1,1130 @@
...
+            "$ref": "./examples/ServerCreateReplica.json"
+          },
+          "Create a server with byok": {
+            "$ref": "./examples/ServerCreateWithBYOK.json"
+          }
+        },
+        "description": "Creates a new server or updates an existing server. The update action will overwrite the existing server.",
+        "parameters": [
+          {
+            "$ref": "../../../../../common-types/resource-management/v1/types.json../../../../../common-types/resource-management/v1/types.json#/parameters/ApiVersionParameter"
```
Comment (raosuhas) [inline specification/mysql/resource-manager/Microsoft.DBforMySQL/preview/2021-12-01-preview/FlexibleServers.json:60]: >s.json../../../../../common-types/resource-man [](http://example.com/codeflow?start=76&length=46)

fix path

File: specification/mysql/resource-manager/Microsoft.DBforMySQL/preview/2021-12-01-preview/FlexibleServers.json [hunk 2]
```diff
@@ -0,0 +1,1130 @@
...
+              "$ref": "#/definitions/Server"
+            }
+          },
+          "201": {
+            "description": "Created",
+            "schema": {
+              "$ref": "#/definitions/Server"
+            }
+          },
+          "202": {
```
Comment (raosuhas) [inline specification/mysql/resource-manager/Microsoft.DBforMySQL/preview/2021-12-01-preview/FlexibleServers.json:94]: >{ [](http://example.com/codeflow?start=17&length=1)

we dont use the 202 model for PUTs anymore : 

[https://github.com/Azure/azure-resource-manager-rpc/blob/master/v1.0/async-api-reference.md#creatingupdating-using-put](https://github.com/Azure/azure-resource-manager-rpc/blob/master/v1.0/async-api-reference.md#creatingupdating-using-put)

File: specification/mysql/resource-manager/Microsoft.DBforMySQL/preview/2021-12-01-preview/FlexibleServers.json [hunk 3]
```diff
@@ -0,0 +1,1130 @@
...
+            "$ref": "./examples/ServerUpdateWithCustomerMaintenanceWindow.json"
+          },
+          "Update server with byok": {
+            "$ref": "./examples/ServerUpdateWithBYOK.json"
+          }
+        },
+        "description": "Updates an existing server. The request body can contain one to many of the properties present in the normal server definition.",
+        "parameters": [
+          {
+            "$ref": "../../../../../common-types/resource-management/v1/types.json../../../../../common-types/resource-management/v1/types.json#/parameters/ApiVersionParameter"
```
Comment (raosuhas) [inline specification/mysql/resource-manager/Microsoft.DBforMySQL/preview/2021-12-01-preview/FlexibleServers.json:125]: >../../../../../ [](http://example.com/codeflow?start=82&length=15)

fix path everywhere in this PR

File: specification/mysql/resource-manager/Microsoft.DBforMySQL/preview/2021-12-01-preview/FlexibleServers.json [hunk 4]
```diff
@@ -0,0 +1,1130 @@
...
+            "required": true,
+            "schema": {
+              "$ref": "#/definitions/ServerForUpdate"
+            },
+            "description": "The required parameters for updating a server."
+          }
+        ],
+        "responses": {
+          "200": {
+            "description": "OK",
```
Comment (raosuhas) [inline specification/mysql/resource-manager/Microsoft.DBforMySQL/preview/2021-12-01-preview/FlexibleServers.json:148]: >OK [](http://example.com/codeflow?start=28&length=2)

do you support both sync and async modes here ?

File: specification/mysql/resource-manager/Microsoft.DBforMySQL/preview/2021-12-01-preview/FlexibleServers.json [hunk 5]
```diff
@@ -0,0 +1,1130 @@
...
+          },
+          {
+            "$ref": "../../../../../common-types/resource-management/v1/types.json#/parameters/ResourceGroupNameParameter"
+          },
+          {
+            "$ref": "#/parameters/ServerNameParameter"
+          }
+        ],
+        "responses": {
+          "200": {
```
Comment (raosuhas) [inline specification/mysql/resource-manager/Microsoft.DBforMySQL/preview/2021-12-01-preview/FlexibleServers.json:191]: >{ [](http://example.com/codeflow?start=17&length=1)

here again do you support both sync and async ? Youonly need to add the ones you actually support here

Comment (raosuhas) [reply specification/mysql/resource-manager/Microsoft.DBforMySQL/preview/2021-12-01-preview/FlexibleServers.json:191]: Same comment everywhere else in this PR

File: specification/mysql/resource-manager/Microsoft.DBforMySQL/preview/2021-12-01-preview/FlexibleServers.json [hunk 6]
```diff
@@ -0,0 +1,1130 @@
...
+            "description": "OK",
+            "schema": {
+              "$ref": "#/definitions/ServerListResult"
```
Comment (raosuhas) [inline specification/mysql/resource-manager/Microsoft.DBforMySQL/preview/2021-12-01-preview/FlexibleServers.json:545]: >ServerListResult [](http://example.com/codeflow?start=37&length=16)

rename to replicalist

Comment (raosuhas) [reply specification/mysql/resource-manager/Microsoft.DBforMySQL/preview/2021-12-01-preview/FlexibleServers.json:545]: Is there only list for the replicas ? What about point Gets \Puts\Deletes for this resource ? How is it created etc ?

File: specification/mysql/resource-manager/Microsoft.DBforMySQL/preview/2021-12-01-preview/FlexibleServers.json [hunk 7]
```diff
@@ -0,0 +1,1130 @@
...
+          }
+        }
+      }
+    },
+    "MaintenanceWindow": {
+      "type": "object",
+      "description": "Maintenance window of a server.",
+      "properties": {
+        "customWindow": {
+          "type": "string",
```
Comment (raosuhas) [inline specification/mysql/resource-manager/Microsoft.DBforMySQL/preview/2021-12-01-preview/FlexibleServers.json:638]: Use enyums where applicable

File: specification/mysql/resource-manager/Microsoft.DBforMySQL/preview/2021-12-01-preview/FlexibleServers.json [hunk 8]
```diff
@@ -0,0 +1,1130 @@
...
+          "$ref": "#/definitions/EnableStatusEnum",
+          "readOnly": true,
+          "description": "Whether or not public network access is allowed for this server. Value is 'Disabled' when server has VNet integration."
+        },
+        "delegatedSubnetResourceId": {
+          "type": "string",
+          "description": "Delegated subnet resource id used to setup vnet for a server."
+        },
+        "privateDnsZoneResourceId": {
+          "type": "string",
```
Comment (raosuhas) [inline specification/mysql/resource-manager/Microsoft.DBforMySQL/preview/2021-12-01-preview/FlexibleServers.json:744]: Consider adding linked access checks where appropriate : 

[https://armwiki.azurewebsites.net/authorization/RBACLinkedAccessCheck.html?q=Linked%20access%20check](https://armwiki.azurewebsites.net/authorization/RBACLinkedAccessCheck.html?q=Linked%20access%20check)

File: specification/mysql/resource-manager/Microsoft.DBforMySQL/preview/2021-12-01-preview/Backups.json [hunk 9]
```diff
@@ -0,0 +1,259 @@
...
+          "Backups"
+        ],
+        "operationId": "Backups_Get",
+        "x-ms-examples": {
+          "Get a backup for a server": {
+            "$ref": "./examples/BackupGet.json"
+          }
+        },
+        "description": "List all the backups for a given server.",
+        "parameters": [
```
Comment (raosuhas) [inline specification/mysql/resource-manager/Microsoft.DBforMySQL/preview/2021-12-01-preview/Backups.json:92]: this is a point GET so should only return one baackup

File: specification/mysql/resource-manager/Microsoft.DBforMySQL/preview/2021-12-01-preview/Backups.json [hunk 10]
```diff
@@ -0,0 +1,259 @@
...
+          "nextLinkName": "nextLink"
+        }
+      }
+    }
+  },
+  "definitions": {
+    "ServerBackupProperties": {
+      "type": "object",
+      "properties": {
+        "backupType": {
```
Comment (raosuhas) [inline specification/mysql/resource-manager/Microsoft.DBforMySQL/preview/2021-12-01-preview/Backups.json:175]: >backupType [](http://example.com/codeflow?start=9&length=10)

use enum if applicable

File: specification/mysql/resource-manager/Microsoft.DBforMySQL/preview/2021-12-01-preview/Configurations.json [hunk 11]
```diff
@@ -0,0 +1,481 @@
...
+          },
+          {
+            "$ref": "#/parameters/ConfigurationKeywordParameter"
+          },
+          {
+            "$ref": "#/parameters/ConfigurationPageParameter"
+          },
+          {
+            "$ref": "#/parameters/ConfigurationPageSizeParameter"
+          }
```
Comment (raosuhas) [inline specification/mysql/resource-manager/Microsoft.DBforMySQL/preview/2021-12-01-preview/Configurations.json:232]: I dont see where these params are being used ?

File: specification/mysql/resource-manager/Microsoft.DBforMySQL/preview/2021-12-01-preview/LogFiles.json [hunk 12]
```diff
@@ -0,0 +1,173 @@
...
+      "authorizationUrl": "https://login.microsoftonline.com/common/oauth2/authorize",
+      "flow": "implicit",
+      "description": "Azure Active Directory OAuth2 Flow",
+      "scopes": {
+        "user_impersonation": "impersonate your user account"
+      }
+    }
+  },
+  "paths": {
+    "/subscriptions/{subscriptionId}/resourceGroups/{resourceGroupName}/providers/Microsoft.DBforMySQL/flexibleServers/{serverName}/logFiles": {
```
Comment (raosuhas) [inline specification/mysql/resource-manager/Microsoft.DBforMySQL/preview/2021-12-01-preview/LogFiles.json:37]: >logFiles [](http://example.com/codeflow?start=132&length=8)

you also need to implement point gets for each individual logfile proxy resource. Can you also expland on how these are created \deleted without PUT\DELETE operations ?

File: specification/mysql/resource-manager/Microsoft.DBforMySQL/preview/2021-12-01-preview/ServiceOperations.json [hunk 13]
```diff
@@ -0,0 +1,599 @@
...
+      "flow": "implicit",
+      "description": "Azure Active Directory OAuth2 Flow",
+      "scopes": {
+        "user_impersonation": "impersonate your user account"
+      }
+    }
+  },
+  "paths": {
+    "/subscriptions/{subscriptionId}/providers/Microsoft.DBforMySQL/locations/{locationName}/capabilities": {
+      "get": {
```
Comment (raosuhas) [inline specification/mysql/resource-manager/Microsoft.DBforMySQL/preview/2021-12-01-preview/ServiceOperations.json:38]: >get [](http://example.com/codeflow?start=7&length=3)

is this a list operation ? THen you need to define the cpability as a resource else this should be modelled as a POST i,e POST .../getCapabilities

File: specification/mysql/resource-manager/Microsoft.DBforMySQL/preview/2021-12-01-preview/ServiceOperations.json [hunk 14]
```diff
@@ -0,0 +1,599 @@
...
+              "$ref": "#/definitions/CloudError"
+            }
+          }
+        },
+        "x-ms-pageable": {
+          "nextLinkName": "nextLink"
+        }
+      }
+    },
+    "/subscriptions/{subscriptionId}/providers/Microsoft.DBforMySQL/locations/{locationName}/checkVirtualNetworkSubnetUsage": {
```
Comment (raosuhas) [inline specification/mysql/resource-manager/Microsoft.DBforMySQL/preview/2021-12-01-preview/ServiceOperations.json:79]: >checkVirtualNetworkSubnetUsage [](http://example.com/codeflow?start=93&length=30)

I am curious as to why is this modeled off the location and not he virtualnetwork itself. Also how are you able to get the permissions to do operations on the virtuaNetwork ?

File: specification/mysql/resource-manager/Microsoft.DBforMySQL/preview/2021-12-01-preview/ServiceOperations.json [hunk 15]
```diff
@@ -0,0 +1,599 @@
...
+          "default": {
+            "description": "Error response describing why the operation failed.",
+            "schema": {
+              "$ref": "#/definitions/CloudError"
+            }
+          }
+        }
+      }
+    },
+    "/subscriptions/{subscriptionId}/providers/Microsoft.DBforMySQL/locations/{locationName}/checkNameAvailability": {
```
Comment (raosuhas) [inline specification/mysql/resource-manager/Microsoft.DBforMySQL/preview/2021-12-01-preview/ServiceOperations.json:127]: >}/ [](http://example.com/codeflow?start=91&length=2)

Again why is this api off the locations ?IN what case would the user need to check location level name availability ?

File: specification/mysql/resource-manager/Microsoft.DBforMySQL/preview/2021-12-01-preview/FlexibleServers.json (hunk 1 above)
Comment (zehao-ms) [reply specification/mysql/resource-manager/Microsoft.DBforMySQL/preview/2021-12-01-preview/FlexibleServers.json]: Fixed

File: specification/mysql/resource-manager/Microsoft.DBforMySQL/preview/2021-12-01-preview/FlexibleServers.json (hunk 2 above)
Comment (zehao-ms) [reply specification/mysql/resource-manager/Microsoft.DBforMySQL/preview/2021-12-01-preview/FlexibleServers.json:94]: This is a legacy API copied from previous API version.

File: specification/mysql/resource-manager/Microsoft.DBforMySQL/preview/2021-12-01-preview/FlexibleServers.json (hunk 3 above)
Comment (zehao-ms) [reply specification/mysql/resource-manager/Microsoft.DBforMySQL/preview/2021-12-01-preview/FlexibleServers.json]: Fixed

File: specification/mysql/resource-manager/Microsoft.DBforMySQL/preview/2021-12-01-preview/FlexibleServers.json [hunk 16]
```diff
@@ -0,0 +1,1130 @@
...
+        "description": "Creates a new server or updates an existing server. The update action will overwrite the existing server.",
+        "parameters": [
+          {
+            "$ref": "../../../../../common-types/resource-management/v1/types.json../../../../../common-types/resource-management/v1/types.json#/parameters/ApiVersionParameter"
+          },
+          {
+            "$ref": "../../../../../common-types/resource-management/v1/types.json#/parameters/SubscriptionIdParameter"
+          },
+          {
+            "$ref": "../../../../../common-types/resource-management/v1/types.json#/parameters/ResourceGroupNameParameter"
```
[the remark this answers is not in the document]
Comment (zehao-ms) [reply specification/mysql/resource-manager/Microsoft.DBforMySQL/preview/2021-12-01-preview/FlexibleServers.json:66]: Copied from previous API version

File: specification/mysql/resource-manager/Microsoft.DBforMySQL/preview/2021-12-01-preview/FlexibleServers.json (hunk 4 above)
Comment (zehao-ms) [reply specification/mysql/resource-manager/Microsoft.DBforMySQL/preview/2021-12-01-preview/FlexibleServers.json:148]: Copied from previous API version

File: specification/mysql/resource-manager/Microsoft.DBforMySQL/preview/2021-12-01-preview/FlexibleServers.json (hunk 5 above)
Comment (zehao-ms) [reply specification/mysql/resource-manager/Microsoft.DBforMySQL/preview/2021-12-01-preview/FlexibleServers.json:191]: Copied from previous API version

File: specification/mysql/resource-manager/Microsoft.DBforMySQL/preview/2021-12-01-preview/FlexibleServers.json (hunk 6 above)
Comment (zehao-ms) [reply specification/mysql/resource-manager/Microsoft.DBforMySQL/preview/2021-12-01-preview/FlexibleServers.json:545]: Copied from previous API version.
The replica is a kind of special server. They are created by creating server API with parameters, so there is only list for this.

File: specification/mysql/resource-manager/Microsoft.DBforMySQL/preview/2021-12-01-preview/FlexibleServers.json (hunk 7 above)
Comment (zehao-ms) [reply specification/mysql/resource-manager/Microsoft.DBforMySQL/preview/2021-12-01-preview/FlexibleServers.json:638]: Copied from previous API version.

File: specification/mysql/resource-manager/Microsoft.DBforMySQL/preview/2021-12-01-preview/FlexibleServers.json [hunk 17]
```diff
@@ -0,0 +1,1130 @@
...
+        }
+      },
+      "description": "Storage Profile properties of a server"
+    },
+    "Sku": {
+      "type": "object",
+      "properties": {
+        "name": {
+          "type": "string",
+          "description": "The name of the sku, e.g. Standard_D32s_v3."
```
[the remark this answers is not in the document]
Comment (zehao-ms) [reply specification/mysql/resource-manager/Microsoft.DBforMySQL/preview/2021-12-01-preview/FlexibleServers.json:709]: Copied from previous API version.

File: specification/mysql/resource-manager/Microsoft.DBforMySQL/preview/2021-12-01-preview/FlexibleServers.json (hunk 8 above)
Comment (zehao-ms) [reply specification/mysql/resource-manager/Microsoft.DBforMySQL/preview/2021-12-01-preview/FlexibleServers.json:744]: Copied from previous API version.

File: specification/mysql/resource-manager/Microsoft.DBforMySQL/preview/2021-12-01-preview/FlexibleServers.json [hunk 18]
```diff
@@ -0,0 +1,1130 @@
...
+        },
+        "maxFailoverSeconds": {
+          "type": "integer",
+          "format": "int32",
+          "description": "The maximum allowed failover time in seconds."
+        }
+      },
+      "description": "Server restart parameters."
+    },
+    "Identity": {
```
[the remark this answers is not in the document]
Comment (zehao-ms) [reply specification/mysql/resource-manager/Microsoft.DBforMySQL/preview/2021-12-01-preview/FlexibleServers.json:1051]: Copied from previous API version.

File: specification/mysql/resource-manager/Microsoft.DBforMySQL/preview/2021-12-01-preview/Backups.json (hunk 9 above)
Comment (zehao-ms) [reply specification/mysql/resource-manager/Microsoft.DBforMySQL/preview/2021-12-01-preview/Backups.json:92]: Fixed description

File: specification/mysql/resource-manager/Microsoft.DBforMySQL/preview/2021-12-01-preview/Backups.json [hunk 19]
```diff
@@ -0,0 +1,259 @@
...
+      "flow": "implicit",
+      "description": "Azure Active Directory OAuth2 Flow",
+      "scopes": {
+        "user_impersonation": "impersonate your user account"
+      }
+    }
+  },
+  "paths": {
+    "/subscriptions/{subscriptionId}/resourceGroups/{resourceGroupName}/providers/Microsoft.DBforMySQL/flexibleServers/{serverName}/backups/{backupName}": {
+      "put": {
```
[the remark this answers is not in the document]
Comment (zehao-ms) [reply specification/mysql/resource-manager/Microsoft.DBforMySQL/preview/2021-12-01-preview/Backups.json:38]: Copied from previous API version.
Yes, currently backups are managed by service and cannot be deleted by customer.

File: specification/mysql/resource-manager/Microsoft.DBforMySQL/preview/2021-12-01-preview/Backups.json (hunk 10 above)
Comment (zehao-ms) [reply specification/mysql/resource-manager/Microsoft.DBforMySQL/preview/2021-12-01-preview/Backups.json:175]: Copied from previous API version.

File: specification/mysql/resource-manager/Microsoft.DBforMySQL/preview/2021-12-01-preview/Configurations.json [hunk 20]
```diff
@@ -0,0 +1,481 @@
...
+            "description": "Error response describing why the operation failed.",
+            "schema": {
+              "$ref": "#/definitions/CloudError"
+            }
+          }
+        }
+      }
+    },
+    "/subscriptions/{subscriptionId}/resourceGroups/{resourceGroupName}/providers/Microsoft.DBforMySQL/flexibleServers/{serverName}/updateConfigurations": {
+      "post": {
```
[the remark this answers is not in the document]
Comment (zehao-ms) [reply specification/mysql/resource-manager/Microsoft.DBforMySQL/preview/2021-12-01-preview/Configurations.json:139]: Copied from previous API version.
These configurations are a set of parameters of MySQL itself.
The service exposes some parameters to customer for modifying and there is no resource created on this.

File: specification/mysql/resource-manager/Microsoft.DBforMySQL/preview/2021-12-01-preview/Configurations.json (hunk 11 above)
Comment (zehao-ms) [reply specification/mysql/resource-manager/Microsoft.DBforMySQL/preview/2021-12-01-preview/Configurations.json:232]: They are used in Configurations_ListByServer

File: specification/mysql/resource-manager/Microsoft.DBforMySQL/preview/2021-12-01-preview/LogFiles.json (hunk 12 above)
Comment (zehao-ms) [reply specification/mysql/resource-manager/Microsoft.DBforMySQL/preview/2021-12-01-preview/LogFiles.json:37]: A legacy API in previous version.
Logfiles are some logs generated by service program on backend. They are managed by service, so there is no PUT/DELETE

File: specification/mysql/resource-manager/Microsoft.DBforMySQL/preview/2021-12-01-preview/ServiceOperations.json (hunk 13 above)
Comment (zehao-ms) [reply specification/mysql/resource-manager/Microsoft.DBforMySQL/preview/2021-12-01-preview/ServiceOperations.json:38]: Copied from previous API version.

File: specification/mysql/resource-manager/Microsoft.DBforMySQL/preview/2021-12-01-preview/ServiceOperations.json (hunk 14 above)
Comment (zehao-ms) [reply specification/mysql/resource-manager/Microsoft.DBforMySQL/preview/2021-12-01-preview/ServiceOperations.json:79]: Copied from previous API version.
The virtualNetwork here should mean the Vnet related settings on this server.

File: specification/mysql/resource-manager/Microsoft.DBforMySQL/preview/2021-12-01-preview/ServiceOperations.json (hunk 15 above)
Comment (zehao-ms) [reply specification/mysql/resource-manager/Microsoft.DBforMySQL/preview/2021-12-01-preview/ServiceOperations.json:127]: Copied from previous API version.

**Comment by zizw123** [comment]
@zehao-ms I see a lot of comments saying it's copied from previous version, but in the change I don't see the reference for the previous version. Can you make sure you put the previous version in the first commit so we can use it view the diff? 

This is explained in detail here (If using Open API which is recommended) https://dev.azure.com/azure-sdk/internal/_wiki/wikis/internal.wiki/208/OpenAPI-Hub-Adding-new-API-version or here(For manual process) : https://dev.azure.com/azure-sdk/internal/_wiki/wikis/internal.wiki/83/Manual-Process-Adding-new-API-version

**Comment by zehao-ms** [comment]
Open a new PR by openAI hub:
https://github.com/Azure/azure-rest-api-specs/pull/19454

**Pull request closed without merging**